File: Camera.InMemoryCamera/CameraService.vb
Imports Core.Extensions

Public Class CameraService
    Implements ICameraService
    Private _currentTarget As Guid
    Private _lastTarget As Guid

    Public ReadOnly Property CurrentTarget As Guid Implements ICameraService.CurrentTarget
        Get
            Return _currentTarget
        End Get
    End Property

    Public ReadOnly Property LastTarget As Guid Implements ICameraService.LastTarget
        Get
            Return _lastTarget
        End Get
    End Property

    Public Sub SetNewTarget(target As Guid) Implements ICameraService.SetNewTarget
        If Not target.IsEmpty() Then
            _lastTarget = _currentTarget
            _currentTarget = target
        End If
    End Sub

End Class

File: space-game/Camera.InMemoryCamera/CameraService.cs
using Core.Extensions;
using System;

namespace Camera.InMemoryCamera
{
    public class CameraService : ICameraService
    {
        public Guid CurrentTarget { get; private set; }

        public Guid LastTarget { get; private set; }

        public void SetTarget(string target)
        {
            if (!string.IsNullOrWhiteSpace(target))
            {
                Guid targetGuid;
                if (Guid.TryParse(target, out targetGuid))
                {
                    SetTarget(targetGuid);
                }
            }
        }

        public void SetTarget(Guid target)
        {
            if (!target.IsEmpty())
            {
                LastTarget = CurrentTarget;
                CurrentTarget = target;
            }
        }

    }
}
